src/signalrclient/hub_connection_impl.cpp

```diff
@@ -251,6 +336,7 @@ namespace signalr
                     if (found != obj.end())
                     {
                         m_handshakeTask->set(std::make_exception_ptr(signalr_exception(std::string("Received unexpected message while waiting for the handshake response."))));
+                        return;
```

Comment:
Let's write a test for this verifying `on` callbacks don't get called in this case now.